android/titanium/src/java/org/appcelerator/titanium/TiBaseActivity.java

```diff
@@ -1615,8 +1602,10 @@ protected void onDestroy()
 		// If the activity is finishing, remove the windowId and supportHelperId so the window and supportHelper can be released.
 		// If the activity is forced to destroy by Android OS, keep the windowId and supportHelperId so the activity can be recovered.
 		if (isFinishing) {
-			int windowId = getIntentInt(TiC.INTENT_PROPERTY_WINDOW_ID, -1);
-			TiActivityWindows.removeWindow(windowId);
+			if (this.launchIntent != null) {
+				int windowId = this.launchIntent.getIntExtra(TiC.INTENT_PROPERTY_WINDOW_ID, -1);
+				TiActivityWindows.removeWindow(windowId);
```

Comment:
I was going to tag this that we should guard for invalid indices, but I see that it uses a SparseArray under the hood - so presumably it "handles" invalid indices by doing nothing?

Reply:
I thought about doing that too, but decided to maintain the old behavior since I've never mucked with our `TiActivityWindows` handling behavior. I can re-evaluate that side of it later.